src/utils/update/review.beneficial.owner.ts

```diff
@@ -0,0 +1,98 @@
+import { BeneficialOwnerGov, BeneficialOwnerGovKey } from "../../model/beneficial.owner.gov.model";
+import * as config from "../../config";
+import { ApplicationData } from "../../model";
+import { BeneficialOwnerIndividual, BeneficialOwnerIndividualKey } from "../../model/beneficial.owner.individual.model";
+
+const AllBoTypes = {
+  boiReview: "review_beneficial_owners_individual",
+  boGovReview: "review_beneficial_owners_government_or_public_authority",
+  boIndividual: BeneficialOwnerIndividualKey,
+  boGov: BeneficialOwnerGovKey
+};
+
+const beneficialOwnerIndividualReviewRedirectUrl = `${config.UPDATE_AN_OVERSEAS_ENTITY_URL
+      + config.UPDATE_REVIEW_BENEFICIAL_OWNER_INDIVIDUAL_PAGE
+      + config.REVIEW_BENEFICIAL_OWNER_INDEX_PARAM}`;
+
+const beneficialOwnerGovReviewRedirectUrl = `${config.UPDATE_AN_OVERSEAS_ENTITY_URL
+        + config.UPDATE_REVIEW_BENEFICIAL_OWNER_GOV_PAGE
+        + config.REVIEW_BENEFICIAL_OWNER_INDEX_PARAM}`;
+
+const checkBoIndividualValidation = (boi: BeneficialOwnerIndividual): boolean => {
+  return boi?.usual_residential_address ? true : false;
+};
+
+const checkBoGovValidation = (boGov: BeneficialOwnerGov): boolean => {
+  return boGov?.principal_address ? true : false;
+};
+
+const checkForBackButtonBo = (appData: ApplicationData, boType: string, boRedirectUrl: string) => {
+  const boLength: number = appData[boType]?.length || 0;
+  const boIndex = boLength - 1;
+  const isAppDataAndBoLength = (appData[boType] && boLength >= 1);
+
+  if (isAppDataAndBoLength && (boType === AllBoTypes.boIndividual) && (!checkBoIndividualValidation(appData[boType][boIndex]))
+        ||
+        isAppDataAndBoLength && (boType === AllBoTypes.boGov) && (!checkBoGovValidation(appData[boType][boIndex]))
+  ) {
+    return `${boRedirectUrl}${boIndex}`;
+  }
+};
+
+export const checkAndReviewBeneficialOwner = (appData: ApplicationData): string => {
+  let redirectUrl = "";
+
+  for (const updateAppData in appData.update){
+    switch (updateAppData){
+        case AllBoTypes.boiReview: {
+          const boiFromBackButton = checkForBackButtonBo(appData, AllBoTypes.boIndividual, beneficialOwnerIndividualReviewRedirectUrl);
+          if (boiFromBackButton) {
+            redirectUrl = boiFromBackButton;
+            return redirectUrl;
+          }
```

Comment:
Is ch_reference being set for the reviewed BOs?

Reply:
ch_reference is not set in the controller but parsed from the html page as part of the request body which then gets saved with "setbeneficialownerdata". Please see the html page for both Bo individual and gov